reinforcement_learning/rlclientlib/logger/event_batcher.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#include "event_queue.h"
+#include "ranking_event.h"
+#include "utility/data_buffer.h"
+
+namespace reinforcement_learning {
+  namespace utility {
+    class event_batcher {
+    public:
+      event_batcher();
+      void batch_serialize(data_buffer& oss, size_t& remaining, event_queue<ranking_event>& queue, size_t _send_high_water_mark);
```

Comment:
Can we make it as either single template method or single method within template class here?

Reply:
I spend most of my time yesterday on this.  The problem is that the logic need to know the type of the event so It can call different flatbuffer object builders. Not sure how we can do that in one method.

---
In reply to: [228610394](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1653#discussion_r228610394) [](ancestors = 228610394)